src/build.py

```diff
@@ -788,23 +791,37 @@ def CMakeCommandBase():
     command.append('-DCMAKE_EXPORT_COMPILE_COMMANDS=ON')
     command.append('-DCMAKE_BUILD_TYPE=Release')
     if IsMac():
-        # Target macOS Seirra (10.12)
-        command.append('-DCMAKE_OSX_DEPLOYMENT_TARGET=10.12')
+        command.append('-DCMAKE_OSX_DEPLOYMENT_TARGET=%s' % OSX_TARGET_VERSION)
     elif IsWindows():
         # CMake's usual logic fails to find LUCI's git on Windows
         git_exe = proc.Which('git')
         command.append('-DGIT_EXECUTABLE=%s' % git_exe)
     return command
 
 
-def CMakeCommandNative(args):
+def CMakeCommandNative(args, build_dir):
     command = CMakeCommandBase()
     command.append('-DCMAKE_INSTALL_PREFIX=%s' % GetInstallDir())
     if IsLinux() and host_toolchains.ShouldUseSysroot():
         command.append('-DCMAKE_SYSROOT=%s' % GetPrebuilt(LINUX_SYSROOT))
         command.append('-DCMAKE_EXE_LINKER_FLAGS=-static-libstdc++')
         command.append('-DCMAKE_SHARED_LINKER_FLAGS=-static-libstdc++')
 
+    elif IsMac() and host_toolchains.ShouldUseSysroot():
```

Comment:
Don't mac builds basically always use a sysroot, that comes with the SDK? What would !ShouldUseSysroot mean?

Reply:
I just copied it from the IsLinux case. I can remove the ShouldUseSysroot().

Reply:
Actually since we're passing this explicit flag to CMake we should probably keep it behind `ShouldUseSysroot`. Mostly I'm still trying to fully  understand what this change actually does.

Reply:
In crbug.com/1070150, the Mac build fails due to an absolute path for --sysroot, which is not supported by the Goma backend. To work around that, we create a symlink under the build root to the actual sysroot path, so that it looks like a relative path. We pass in args to cmake to set up the build to use that relative sysroot path.